gst/nnstreamer/tensor_query/nnstreamer_edge_internal.c

```diff
@@ -43,37 +203,108 @@ nns_edge_handle_is_valid (nns_edge_h edge_h)
+_nns_edge_check_connection (nns_edge_conn_s * conn)
+{
+  size_t size;
+  GIOCondition condition;
+
+  condition = g_socket_condition_check (conn->socket,
+      G_IO_IN | G_IO_PRI | G_IO_ERR | G_IO_HUP);
+  size = g_socket_get_available_bytes (conn->socket);
+
+  if (condition && size <= 0) {
+    nns_edge_logw ("Socket is not available, possibly EOS.");
+    return false;
+  }
+
+  return true;
+}
+
 /**
  * @brief Check network connection.
  */
 bool
 nns_edge_is_connected (nns_edge_h edge_h)
 {
-  if (!nns_edge_handle_is_valid (edge_h)) {
-    nns_edge_loge ("Invalid param, given edge handle is invalid.");
-    return false;
+  nns_edge_conn_data_s *conn_data;
+  GList *conn_list = NULL;
+  bool ret = true;
+  UNUSED (edge_h);
+
+  conn_list = g_hash_table_get_values (_table);
+  while (conn_list) {
+    conn_data = (nns_edge_conn_data_s *) conn_list->data;
+    if (false == _nns_edge_check_connection (conn_data->src_conn) ||
+        false == _nns_edge_check_connection (conn_data->sink_conn)) {
+      nns_edge_loge ("Socket is not available.");
+      ret = false;
+      break;
+    }
+    conn_list = g_list_next (conn_list);
   }
+  g_list_free (conn_list);
+
+  return ret;
+}
+
+/**
+ * @brief Get socket address
+ */
+static bool
+_nns_edge_get_saddr (const char *ip, uint16_t port,
+    GCancellable * cancellable, GSocketAddress ** saddr)
+{
+  GError *err = NULL;
+  GInetAddress *addr;
+
+  /* look up name if we need to */
+  addr = g_inet_address_new_from_string (ip);
+  if (!addr) {
+    GList *results;
+    GResolver *resolver;
+    resolver = g_resolver_get_default ();
+    results = g_resolver_lookup_by_name (resolver, ip, cancellable, &err);
+    if (!results) {
+      if (g_error_matches (err, G_IO_ERROR, G_IO_ERROR_CANCELLED)) {
+        nns_edge_logd ("gst_tensor_query_socket_new: Cancelled name resolval");
+      } else {
+        nns_edge_loge ("Failed to resolve ip '%s': %s", ip, err->message);
+      }
+      g_clear_error (&err);
+      g_object_unref (resolver);
+      return false;
+    }
+    /** @todo Try with the second address if the first fails */
+    addr = G_INET_ADDRESS (g_object_ref (results->data));
+    g_resolver_free_addresses (results);
+    g_object_unref (resolver);
+  }
+
+  *saddr = g_inet_socket_address_new (addr, port);
+  g_object_unref (addr);
 
-  /** @todo check connection status */
   return true;
 }
 
 /**
  * @brief Get registered handle. If not registered, create new handle and register it.
  */
 int
-nns_edge_get_handle (const char *id, const char *topic, nns_edge_h * edge_h)
+nns_edge_create_handle (const char *id, const char *topic, nns_edge_h * edge_h)
 {
   nns_edge_handle_s *eh;
 
   if (!id || *id == '\0') {
     nns_edge_loge ("Invalid param, given ID is invalid.");
     return NNS_EDGE_ERROR_INVALID_PARAMETER;
   }
-
   if (!topic || *topic == '\0') {
-    nns_edge_loge ("Invalid param, given topic is invalid.");
-    return NNS_EDGE_ERROR_INVALID_PARAMETER;
+    nns_edge_logw
```

Comment:
Q: the case if topic is null or empty.
Why did you change the condition about invalid topic string?